packages/gnarly-core/src/ReducerRunner.ts

```diff
@@ -64,34 +63,46 @@ class ReducerRunner {
           // we're resuming, so replay from store if possible
           try {
             const latestTransaction = await globalState.store.getLatestTransaction(this.reducer.config.key)
-            if (!latestTransaction) {
-              throw new Error('No latest transaction available.')
-            }
 
-            // we're resuming, so reload blockstreamer chain
-            historicalBlocks = await globalState.store.getHistoricalBlocks(this.reducer.config.key)
+            // load historical chain
+            const historicalBlocks = await globalState.store.getHistoricalBlocks(this.reducer.config.key)
 
-            // if no history is available, resume from blockHash
-            if (!historicalBlocks.length) {
-              latestBlockHash = latestTransaction.blockHash
+            if (!latestTransaction || !historicalBlocks || historicalBlocks.length === 0) {
+              throw new Error('No latest transaction or historical blocks available, skipping resumption.')
             }
 
             try {
+              const mostRecentHistoricalBlock = historicalBlocks[historicalBlocks.length - 1]
+
+              assert.equal(
```

Comment:
lets say last transaction was at block 100
and we let the reducer run until the saved chain has blocks from 101 - 201

this is only okay if getLatestTransaction includes historicalblocks transactions, if it does not include historicalblocks transactions then there is a possibility that the last reducer transaction was done 100 blocks ago

Reply:
lastTransaction should always be the most recent `Ourbit#Transaction`, which is produced after a block is found, just like historical blocks. The only time this won't be true is if the process crashing right in the middle of the part where we `commitTransaction` and `saveHistoricalBlock`, right? Maybe I'm misunderstanding, sorry

Reply:
so if I understand correctly, Ourbit#Transaction will be produced every block since we save the historical block into the db?

Reply:
An `Ourbit#Transaction` is the set of all of the patches and operations that were performed in one atomic set (like a blockstream block). So it _is_ produced for every block (after that block has been processed), but not because of the historical blocks; it's job is to track the patches so that we can potentially revert them later.